scipy/stats/_kde.py

```diff
@@ -574,17 +577,25 @@ def _compute_covariance(self):
         covariance_factor().
         """
         self.factor = self.covariance_factor()
-        # Cache covariance and inverse covariance of the data
-        if not hasattr(self, '_data_inv_cov'):
+        # Cache covariance and permuted cholesky decomp of permuted covariance
+        if not hasattr(self, '_data_cho_cov'):
             self._data_covariance = atleast_2d(cov(self.dataset, rowvar=1,
                                                bias=False,
                                                aweights=self.weights))
-            self._data_inv_cov = linalg.inv(self._data_covariance)
+            self._data_cho_cov = linalg.cholesky(
+                self._data_covariance[::-1, ::-1]).T[::-1, ::-1]
```

Comment:
I think there's enough going on here that there should be some comments to explain things. It took me a bit to figure out what's happening. 

Just to make I'm on the same page, here are the details as I understand them:

<details>
Let $\Gamma$ be the covariance matrix for the Gaussian kernel and $LL^{\top} = \Gamma^{-1}$ be the Cholesky decomposition of the precision matrix $\Gamma^{-1}$. Later on, we want to be able to transform $n \times d$ data matrices $X$ by multiplying on the right by the Cholesky factor $L$ for $\Gamma^{-1}$. Equivalently,  we want to be able to find $Z$ such that $Z = XL$.

If we know $L^{-1}$ we can instead use a triangular solver to $ZL^{-1} = X$. It turns out we can calculate $L^{-1}$ with a Cholesky transform, but not of $\Gamma$, but instead a permuted version of $\Gamma$. 

If $RR^{\top} = \Gamma$ then $(R^{-1})^{\top}R^{-1} = \Gamma^{-1}$. This isn't quite a Cholesky decomposition though, since $(R^{-1})^{\top}$ is upper triangular instead of lower triangular. Let $J$ be the antidiagonal matrix with all ones on the antidiagonal and zeros elsewhere. Multiplying on the left by $J$ permutes rows and multiplying on the right by $J$ permutes columns. If we instead calculate the Cholesky decomposition $RR^{\top} = J\Gamma J$, then
$$(R^{-1})^{\top}R^{-1} = J^{-1}\Gamma^{-1}J^{-1} = J\Gamma^{-1}J$$
and thus
$$\Gamma^{-1} = J(R^{-1})^\top R^{-1}J = (JR^{-1}J)^\top(JR^{-1}J)$$

where we've used that $J = J^{-1}$ and $J = J^{\top}$. 

This means that if we know the Cholesky factor $R$ of $J\Gamma J$, the Cholesky factor of $\Gamma^{-1}$ is 
$J(R^{-1})^{\top}J$. This means $L^{-1} = JR^{\top}J$, and we can write our equation as $ZJR^{\top}J = X$ and solve for $Z$. This is exactly what you've done, but it's still not intuitive for me yet, just algebraic.
</details>

I think we should have a comment explaining the types of equations we want to be able to solve
$Z = XL$, where $L = \operatorname{Chol}\left(\Gamma^{-1}\right)$. Equivalently $ZL^{-1} = X$. Also a brief sentence explaining that it's possible to calculate $L^{-1}$ directly from the Cholesky decompostion of the permuted matrix that reverses both the rows and columns of $X$. Also some kind of citation to a place to find more details. The best explanation I've found is in a [Mathoverflow answer](https://mathoverflow.net/a/230817) by the eminent mathematician Robert Israel. Perhaps just a link to this answer would be good enough.

Reply:
It definitely does deserve an explanation, and I meant to include one. Sorry to make you find it on your own! Yes, I think that is the original post I followed. I'll write a bit about the motivation and link to that.

Reply:
No worries. The explanation is very clear now. I think everything is in good shape now but still want to double check carefully.

Reply:
I didn't follow the whole argument but if applies here, use the `lower=False` keyword for starting with an upper triangular in calling `cholesky`. Might save a column swap or two.

Reply:
> I didn't follow the whole argument but if applies here, use the `lower=False` keyword for starting with an upper triangular in calling `cholesky`. Might save a column swap or two.

It's a good thought but isn't quite what we want. According to the documentation for the [underlying Lapack function](https://www.math.utah.edu/software/lapack/lapack-d/dpotrf.html),  if $LL^{\top}$ is the Cholesky factorization of $\Gamma$, then
 `cholesky(Gamma, lower=False)` will return `L` and `cholesky(Gamma, lower=True)` will return $L^{\top}$. I've tried it be sure. What we would actually need is to find an upper triangular matrix $U$ such that $\Gamma = UU^{\top}$. The  `cholesky` function can't do this so we're required to do the trick with reversing the rows and columns.

Reply:
Oh, but I guess it will save us one transpose when computing the Cholesky decomposition of $J\Gamma J$ since we need the upper triangular factor there. It may be worthwhile for small matrices.